viam-cartographer/scripts/setup_cartographer_macos.sh

```diff
@@ -1,8 +1,9 @@
 #!/bin/bash
 echo "Installing cartographer external dependencies"
 brew update
-brew install abseil boost ceres-solver protobuf ninja cairo googletest lua@5.3 pkg-config cmake go@1.20 grpc 
+brew install abseil boost ceres-solver protobuf@21 ninja cairo googletest lua@5.3 pkg-config cmake go@1.20 grpc 
```

Comment:
yeah, for whatever reason we get errors when using protobuf@23(latest). I can't find the reason why we are getting these errors, and it does not appear that any of our other dependencies require protobuf@21 to work. But switching to protobuf@21 fixes the error

rdk does not use protobuf anymore, so we shouldn't have issues with that repo. we will still have issues with the app and api repos

Reply:
I'm okay with putting this fix in for now, but I'm a bit worried about pinning to a specific version of protobuf longterm since it will get older and older. Mind asking James to sign off on this one as well?